src/Microsoft.AspNet.Server.Kestrel/Http/SocketOutput.cs

```diff
@@ -312,10 +302,6 @@ private void OnWriteCompleted(WriteContext writeContext)
                 {
                     scheduleWrite = true;
```

Comment:
There might be a way to do something like that. The key is to always do shutdown and disconnect in the right order and to never do either twice.